src/BloomExe/Edit/PageEditingModel.cs

```diff
@@ -12,7 +14,8 @@ public class PageEditingModel
 	{
 		public void ChangePicture(string bookFolderPath, GeckoHtmlElement img, PalasoImage imageInfo, IProgress progress)
 		{
-			var imageFileName = ImageUtils.ProcessAndSaveImageIntoFolder(imageInfo, bookFolderPath);
```

Comment:
In PageEditingModel.ChangePicture(), a check is now made for whether the chosen picture file is actually the same as the original picture file referenced in the book's HTML.  If it is, then we want to write it without changing its name, whether or not it has actually changed.  If it isn't the same file (even if the base filename is the same), then when it is written, it has to be renamed if the book folder already has a picture file with the same basename.